C#/C#.sublime-syntax

```diff
@@ -1,278 +1,1634 @@
+
+    # comments
+  comments:
+    - match: '^\s*(///)'
       captures:
-        1: meta.namespace.identifier.source.cs
-        2: keyword.other.namespace.source.cs
-        3: entity.name.type.namespace.source.cs
+        1: punctuation.definition.comment.documentation.cs
+      push: documentation
+    - match: '//'
+      scope: punctuation.definition.comment.cs
       push:
-        - meta_scope: meta.namespace.source.cs
-        - match: "}"
-          scope: punctuation.section.namespace.end.source.cs
+        - meta_scope: comment.line.double-slash.cs
+        - match: $
           pop: true
-        - match: "{"
-          scope: punctuation.section.namespace.begin.source.cs
-          push:
-            - meta_scope: meta.namespace.body.source.cs
-            - match: "(?=})"
-              pop: true
-            - include: code
-    - include: code
-  block:
-    - match: "{"
-      scope: punctuation.section.block.begin.source.cs
+        - include: comments_in
+    - match: '/\*'
+      scope: punctuation.definition.comment.begin.cs
+      push:
+        - meta_scope: comment.block.cs
+        - match: '\*/'
+          scope: punctuation.definition.comment.end.cs
+          pop: true
+        - include: comments_in
+
+  comments_in:
+    - match: \b(?i:todo|hack|xxx)\b
+      scope: comment.line.todo
+    - match: '/!\\'
+      scope: comment.line.todo
+    - match: \b(https?://\S*)\b
+      scope: markup.underline.link.cs
+
+  git_merge:
+    - match: '^(<<<<<<<).*$'
+      captures:
+        1: meta.git.merge punctuation.section.merge.begin.git
+    - match: '^(=======).*$'
+      captures:
+        1: meta.git.merge punctuation.separator.merge.git
+    - match: '^(>>>>>>>).*$'
+      captures:
+        1: meta.git.merge punctuation.section.merge.end.git
+
+  preprocessor_option:
+    - meta_scope: meta.preprocessor.cs
+    - match: '(define|undef)\s+{{name}}'
+      captures:
+        1: keyword.control.preprocessor.cs
+        2: constant.numeric.cs
+    - match: '((el)?if)'
+      scope: keyword.control.preprocessor.cs
       push:
-        - meta_scope: meta.block.source.cs
-        - match: "}"
-          scope: punctuation.section.block.end.source.cs
-          pop: true
-        - include: code
-  builtinTypes:
-    - match: \b(bool|byte|sbyte|char|decimal|double|float|int|uint|long|ulong|object|short|ushort|string|void|class|struct|enum|interface)\b
-      scope: storage.type.source.cs
-  class:
-    - match: '(?=\w?[\w\s]*(?:class|struct|interface|enum)\s+\w+)'
+        - match: '\(|\)'
+          scope: punctuation.definition.expression.cs
+        - match: '&&|\|\||!'
+          scope: keyword.operator.logical.cs
+        - match: '\b(true|false)\b'
+          scope: constant.language.cs
+        - match: '{{name}}'
+          scope: constant.numeric.cs
+        - include: option_done
+    - match: '(else|endif)'
+      scope: keyword.control.preprocessor.cs
+      push: option_done
+    - match: '(error|warning)\s*'
```

Comment:
Add `\b`, and change the `scope:` to
```
captures:
    1:
```
so that the whitespace doesn't get a scope.